src/uint256.h

```diff
@@ -69,16 +69,27 @@ class base_blob
 
     /** @name Hex representation
      *
-     * The reverse-byte hex representation is a convenient way to view the blob
-     * as a number, because it is consistent with the way the base_uint class
-     * converts blobs to numbers.
+     * The hex representation used by GetHex(), ToString(), FromHex() and
+     * SetHexDeprecated() is unusual, since it shows bytes of the base_blob in
+     * reverse order. For example, a 4-byte blob {0x12, 0x34, 0x56, 0x78} is
+     * represented as "78563412" instead of the more typical "12345678"
+     * representation that would be shown in a hex editor or used by typical
+     * byte-array / hex conversion functions like python's bytes.hex() and
+     * bytes.fromhex().
+     *
+     * The nice thing about the reverse-byte representation, even though it is
+     * unusual, is that if a blob contains an arithmetic number in little endian
+     * format (with least significant bytes first, and most significant bytes
+     * last), the GetHex() output will match the way the number would normally
+     * be written in base-16 (with most significant digits first and least
+     * significant digits last).
+     *
+     * This means, for example, that ArithToUint256(num).GetHex() can be used to
+     * used to display an arith_uint256 num value as a number, because
+     * ArithToUint256() converts the number to a blob in little-endian format,
+     * so the arith_uint256 class doesn't need to have its own number parsing
```

Comment:
Could we be more compact here?
Personally I'm not a fan of long documentation - though the reason usually is that if we need to document something heavily it's a sign of poorly written code - not something we can really change here.

Reply:
I think it's a useful comment.

Reply:
'can be used to used to display' -> 'can be used to display'

Reply:
This was already fixed [`13464c0c44645e0ed88d9d72c3ad697dca800e10..3e0a992a3f0f2b15b7be5049dc4f3134b4b0bc40`#diff-800776e2dd](https://github.com/bitcoin/bitcoin/compare/13464c0c44645e0ed88d9d72c3ad697dca800e10..3e0a992a3f0f2b15b7be5049dc4f3134b4b0bc40#diff-800776e2dda39116e889839f69409571a5d397de048a141da7e4003bc099e3e2)L88

Reply:
👍 my bad, github didn't show it as outdated and was looking at the convo tab.